flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DeleteNodeGenerator.java

```diff
@@ -96,9 +95,12 @@ public JsonElement getTextEditsToDeletedNode(Document document, Project project)
         return gson.toJsonTree(textEditsMap);
     }
 
-    private ImportDeclarationNode getUnusedImport(String unusedModule, NodeList<ImportDeclarationNode> imports) {
+    private ImportDeclarationNode getUnusedImport(TextRange diagnosticLocation,
+                                                  NodeList<ImportDeclarationNode> imports) {
         for (ImportDeclarationNode importNode : imports) {
-            if (importNode.moduleName().get(0).text().equals(unusedModule)) {
+            TextRange importNodeLocation = importNode.textRange();
+            if (importNodeLocation.startOffset() <= diagnosticLocation.startOffset() &&
+                    importNodeLocation.endOffset() >= diagnosticLocation.endOffset()) {
```

Comment:
Shall we use `PositionUtil.isWithinRange` here?
https://github.com/ballerina-platform/ballerina-lang/blob/6335c0648658d7857dd6615938ac46dc95e8d407/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/PositionUtil.java#L114

Reply:
Changed